src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs

```diff
@@ -960,6 +960,61 @@ void TestAction(HttpResponse httpResponse)
             Assert.Equal(httpContext.Response, httpResponseArgument);
         }
 
+        [Fact]
+        public async Task RequestDelegateSetContentTypeWhenNullAndReturnTypeIsString()
+        {
+            string Plaintext() => "Hello, World!";
+            var httpContext = new DefaultHttpContext();
+
+            var requestDelegate = RequestDelegateFactory.Create((Func<string>)Plaintext, new EmptyServiceProvider());
+
+            await requestDelegate(httpContext);
+
+            Assert.Equal(ContentTypeConstants.PlainTextContentTypeWithCharset, httpContext.Response.ContentType);
+        }
+
+        [Fact]
+        public async Task RequestDelegateSetContentTypeWhenNullAndReturnTypeIsGenericOfTypeTask()
+        {
+            async Task<string> Plaintext()
+            {
+                var sayHelloTask = Task.Run(() =>
```

Comment:
We should just update the existing `RequestDelegateWritesComplexReturnValueAsJsonResponseBody` (and rename it) test to verify the httpContext.Response.ContentType, but in the future it's a more efficient to use `Task.FromResult` than `Task.Run` when the result is already available. Also, you can just return the Task directly from a non-async method instead of doing a `return await`.